### ISSUE TITLE
Add option to the  'aria-valid-attr' rule to allow specification of custom attributes

### ISSUE THREAD

**Issue by quachtina96** [dequelabs/axe-core#460]
Note: I'm using the version of axe.js built from the shadow DOM branch on Mon Jul 24 12:17:05 2017, with latest revision from commit: d02dba3223fefe525438330e40b5da5de81eeeb5

I have [a Polymer element that has an aria-active-attribute](https://cs.chromium.org/chromium/src/chrome/browser/resources/settings/basic_page/basic_page.html?q=aria-active-attribute&sq=package:chromium&dr=C&l=199) that is used to determine what actual aria attribute to set. axe detects "aria-active-attribute" as an invalid aria attribute name, and I'd like to ignore that warning for elements with that attribute, but the element exists in the shadow dom and include-exclude for shadow DOM is not supported.

![image](https://user-images.githubusercontent.com/10423718/28603241-def60a3a-7177-11e7-9e30-6235ec075004.png)
The attribute is on an element nested in shadow DOM. Since it is part of the Polymer library and not written by me--I thought of asking them to  change the name of the attribute in the next release of Polymer (in about a year) but that would affect backwards compatibility.

Could there be a custom configuration for the `aria-valid-attr` rule to "whitelist" known "bad" ARIA attributes?

**Comment by WilcoFiers** [comment]
You can whitelist attributes by configuring the check, like this:

```js
  axe.configure({
    checks: [{
      id: 'aria-valid-attr',
      options: ['aria-active-attribute']
    }]
  })
```

Check configuration is a bit obscure. We don't have documentation per check on what's configurable, but this is one of them.
